### ISSUE TITLE
distupgrade all + keeporphans + allowuninstall -> deletes orhans

### ISSUE THREAD

**Issue by jsilhan** [openSUSE/libsolv#108]
Hi,
when I distupgrade all packages that doesn't have conflicts it will keep the orphans installed.
So far so good:
```
repo system 0 testtags <inline>
#>=Pkg: A 1 1 x86_64
#>=Pkg: C 1 1 x86_64
repo fedora 0 testtags <inline>
#>=Pkg: A 1 2 x86_64
system x86_64 rpm system
poolflags implicitobsoleteusescolors
solverflags allowvendorchange keepexplicitobsoletes bestobeypolicy keeporphans yumobsoletes
job distupgrade all packages
result transaction,problems <inline>
#>upgrade A-1-1.x86_64@system A-1-2.x86_64@fedora
```

then when I add allowuninstall solver flag to solve conflicts, it will take away orphans too.
(I expect the C to stay installed)
```
repo system 0 testtags <inline>
#>=Pkg: A 1 1 x86_64
#>=Pkg: B 1 1 x86_64
#>=Pkg: C 1 1 x86_64
#>=Pkg: D 1 1 x86_64
repo fedora 0 testtags <inline>
#>=Pkg: A 1 2 x86_64
#>=Pkg: B 1 2 x86_64
#>=Con: D
system x86_64 rpm system
poolflags implicitobsoleteusescolors
solverflags allowvendorchange keepexplicitobsoletes bestobeypolicy keeporphans yumobsoletes allowuninstall
job distupgrade all packages
result transaction,problems <inline>
#>upgrade A-1-1.x86_64@system A-1-2.x86_64@fedora
#>erase D-1-1.x86_64@system
#>upgrade B-1-1.x86_64@system B-1-2.x86_64@fedora
```

Is it intentional? shouldn't allowuninstall job for all packages be the same as allowuninstall solver flag?
This works fine:
```
repo system 0 testtags <inline>
#>=Pkg: A 1 1 x86_64
#>=Pkg: B 1 1 x86_64
#>=Pkg: C 1 1 x86_64
#>=Pkg: D 1 1 x86_64
repo fedora 0 testtags <inline>
#>=Pkg: A 1 2 x86_64
#>=Pkg: B 1 2 x86_64
#>=Con: D
system x86_64 rpm system
poolflags implicitobsoleteusescolors
solverflags allowvendorchange keepexplicitobsoletes bestobeypolicy keeporphans yumobsoletes
job allowuninstall name D
job allowuninstall name C
job distupgrade all packages
result transaction,problems <inline>
#>erase D-1-1.x86_64@system
#>upgrade A-1-1.x86_64@system A-1-2.x86_64@fedora
#>upgrade B-1-1.x86_64@system B-1-2.x86_64@fedora
```
I don't have a problem with is behavior as we can achieve what we want with ^ workaround. I wasn't sure whether it's a bug or not and wondering how does zypper do distro-sync with resolving conflicts and keeping the orphaned packages installed proper way.

**Comment by mlschroe** [comment]
No, it's not intentional. It's allowuninstall interfering with keeporphans. (As for the third testcase working, that's actually another glitch. I forgot one place when I added the allowuninstall job support...)

Ok, all fixed with commit d5af25c0ef8fb01e5b0d377a2d7223281991b960

**Comment by jsilhan** [comment]
You have saved Fedora system upgrades, thanks
